specification/requirements.adoc

```diff
@@ -1,5 +1,167 @@
+physical devices, over DOE mailboxes. CoVE-IO also relies on SPDM for
+gathering devices certificates and measurements.
+
+As a consequence, a CoVE-IO supported DSM must support the SPDM specification
+version 1.2 or above. It must also support sending and receiving SPDM Secured
+Messages as defined in the (<<SecuredSPDM>>) specification version 1.1 or above.
+
+TODO: List of required SPDM messages
+
+===== Integrity and Data Exchange (IDE)
+
+The data that flows between a TDI and a TVM must be confidential and both
+integrity and replay attack protected. PCIe IDE provides this protection for all
+Transport Layer Packets (TLPs) moving between those two endpoints. As a
+consequence, CoVE-IO compatible devices must implement and expose this PCIe
+capability.
+
+====== Selective Streams
+
+As all PCIe switches are excluded from a TVM's TCB, CoVE-IO compliant hosts
+TSMs exclusively establish selective IDE streams between the host PCIe Root Port
+and TEE-IO devices.
+
+====== Key Management
+
+To fully support IDE, CoVE-IO compatible devices must implement the IDE Key
+Management (IDE_KM) flows. IDE_KM messages are initiated by the TSM and sent
+over SPDM to the DSM.
+
+=== Host
+
+In order to support TEE-IO devices with the above described capabilities, a
+CoVE implementation must meet software and hardware requirements, as described
+in the next sections.
+
+==== Ownership
+
+In a CoVE-IO context, the non-confidential host software stack is the sole
+platform resources owner. In particular, the physical devices to which TDIs
+belong are owned and managed by the host, typically through the device PF.
+Host software stack components, e.g. the VMM, assign, unassign and expose
+TDIs resp. to, from and to TVMs.
+
+==== Platform Hardware Components
+
+===== PCIe
+
+All CoVE-IO-compliant devices in a CoVE platform must be connected to the CoVE
+platform through a PCIe Root Port (RP) that is part of a PCIe Root Complex (RC).
+
+A CoVE-IO PCIe link can be direct or go via any topology or PCIe switches and
+bridges. Since those are excluded from a TVM's TCB, only selective IDE streams
+must be used between a CoVE-IO device and its corresponding PCIe Root Port.
+
+The TSM establishes one single selective IDE stream for each physical device
+from which TDIs may be attached to TVMs. All TDIs within a CoVE-IO device share
+the same IDE stream.
+
+For a given selective IDE stream, the TSM generates, owns and distributes the
+IDE keys to both the CoVE-IO device and its upstream PCIe Root Port.
+
+[[coveio_hierarchy]]
+.CoVE-IO PCIe Topology
+image::coveio_hierarchy.svg[align="center"]
+
+
+===== IO Translation Agent
+
+CoVE-IO-compliant platforms must have at least one IO translation agent, also
+known as an IOMMU. Platform IOMMUs must follow the RISC-V <<IOMMU>> architecture
+specification v1.0.
+
+In order to secure direct acces to confidential memory access, all inbound
+traffic from a TDI must be translated IOMMU translated.
```

Comment:
do we need describe the IOMMU capability, such as isolation? Only TSM can touch the trusted translation table, while VMM can only touch the untrusted one.

Reply:
Good point. I added a paragraph for that.